### ISSUE TITLE
[Feature RequesT] Add VPC endpoint to S3

### ISSUE THREAD

**Issue by aledbf** [poseidon/typhoon#579]
### Feature

Add support for a VPC endpoint to S3 https://docs.aws.amazon.com/en_pv/vpc/latest/userguide/vpc-endpoints-s3.html to reduce costs accessing S3 from the cluster

The implementation of this feature is simple https://github.com/aledbf/typhoon/commit/0210e5372b4d8e0d4ae1a86412652003e7a5829b

### Tradeoffs

The commit I posted assumes a VPC endpoint in the same region.

**Comment by dghubble** [comment]
tldr: I'd rather just add an output for the route table id so folks can add these for AWS services as needed.

#### Mini AWS commentary

_Shakes fist at AWS, not you_

So let's say one has an S3 bucket `foo`in `us-west-2` accessed by an app on Kubernetes. For resiliency, that app might run from any cloud or on-prem, from any region (although perhaps one favors `us-west-2`). So the app accesses the bucket by one uniform DNS name [supported by AWS for S3](https://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region) (pick one).

* s3.us-west-2.amazonaws.com
* s3.dualstack.us-west-2.amazonaws.com
* account-id.s3-control.us-west-2.amazonaws.com
* account-id.s3-control.dualstack.us-west-2.amazonaws.com

All good. Now a nice cloud provider might try to improve their service experience when accessed from their own cloud by resolving these names differently for you. GCP makes choices like this automatic. AWS instead invents "VPC Endpoints". Perhaps its a feature that you get manual control. But the con is users get the joy of manually adding a VPC endpoint for every AWS service they might wish to access, for every name it might be accessed by. Apps on a cluster might access `s3.us-west-2.amazonaws.com` and `s3.us-east-1.amazonaws.com` or other AWS services besides S3 too. We'd want to add them all? App details are bleeding into the cluster configuration. And apparently VPC endpoints don't work for the dualstack names. And have their own notions of policy too. Oh, AWS.

#### Conclusion

I think the right course is to enable folks to augment their clusters with VPC endpoint(s) for the service(s) they use, if they feel the network cost savings justify it. Adding an output of the route table id could offer that. That avoids Typhoon getting involved in choosing which AWS services and endpoints to create VPC endpoints for. There may also be some special reason to favor the public service endpoints in certain cases I'm unaware of (which could go to show why AWS took this add each endpoint approach).

Would you be game to add the output and an example that adds it to the route table of a cluster after this section https://typhoon.psdn.io/architecture/aws/#firewalls?

**Comment by dghubble** [comment]
As mentioned, I'd take a route table id output to allow folks to add endpoints to their clusters.

Going this issue about having AWS builtin endpoints.
